boards/posix/nrf52_bsim/Kconfig.defconfig

```diff
@@ -0,0 +1,51 @@
+
+if BOARD_NRF52_BSIM
+
+config BUILD_OUTPUT_BIN
+	default n
+
+config BUILD_OUTPUT_EXE
+	default y
+
+config OUTPUT_PRINT_MEMORY_USAGE
+	default n
+
+config BOARD
+	default "nrf52_bsim"
+
+config SOC_COMPATIBLE_NRF
+	def_bool y
+
+config SOC_COMPATIBLE_NRF52X
+	def_bool y
+
+config SOC_COMPATIBLE_NRF52832
+	def_bool y
```

Comment:
Not super familiar with the details here, but instead of having `Kconfig.defconfig` files override `SOC_COMPATIBLE_*` symbols, maybe they could be `select`ed by other `SOC_*` symbols.

`SOC_FOO_MARK_II` could have `select SOC_COMPATIBLE_FOO` for example, if it's backwards compatible.

Reply:
I did try first to select them in the kconfig.board, but it did not get thru. So I went for the easy path.. 
I guess the problem is that they are defined in the real SOC kconfig files and therefore only included when the real SOC and ARM arch is selected?

Reply:
It's fine if they become undefined for other ARCHes by the way. Undefined symbols evaluate to `n`, and the CI check for undefined symbols (in `scripts/ci/check-compliance.py`) always parses the entire Kconfig tree (it includes all boards and ARCHes by setting `BOARD_DIR` and `ARCH` to glob patterns).

Reply:
Just noticed that the simulated boards use the POSIX arch, so that might not work. Maybe the symbols could be put into `soc/Kconfig` instead. Since non-ARM arches can be compatible with those boards, it'd still make sense from an organizational perspective.

Reply:
> could be put into soc/Kconfig

It would feel funny to me to have that kind of symbol in such a high common place.. Even where they are today, seems a bit to out of place..

Reply:
IMO it makes sense considering that the symbols are arch-independent. Putting them in `soc/Kconfig` makes it explicit.

Having a POSIX board be compatible with an ARM board is pretty unexpected in itself, so it's probably a good thing if it stands out a bit. It makes it easier to notice that something special is going on. :)

(I suspect that a lot of the `Kconfig.defconfig` stuff in Zephyr could be simplified a bit with promptless helper symbols and `select` too, but that's a different issue.)

Maybe it'd be nice to add a comment:

```diff
diff --git a/soc/Kconfig b/soc/Kconfig
index cc6a7ee14..17a3219cb 100644
--- a/soc/Kconfig
+++ b/soc/Kconfig
@@ -11,3 +11,22 @@ osource "$(SOC_DIR)/$(ARCH)/Kconfig"
 osource "$(SOC_DIR)/$(ARCH)/*/Kconfig"
 
 endmenu
+
+# The helper symbols below are put here due to an unusual setup: The simulated
+# nrf52_bsim board uses the POSIX arch, but is compatible with Nordic ARM
+# boards
+
+config SOC_COMPATIBLE_NRF
+       bool
+       help
+         Selected by NRF-compatible boards
+
+config SOC_COMPATIBLE_NRF52X
+       bool
+       help
+         Selected by NRF52X-compatible boards
+
+config SOC_COMPATIBLE_NRF52832
+       bool
+       help
+         Selected by NRF52832-compatible boards
```

Reply:
Anybody disagrees with what Ulf proposes? otherwise I will just push his proposal.